jb/fftw/tde_result.hpp

```diff
@@ -0,0 +1,122 @@
+#ifndef jb_fftw_tde_result_hpp
+#define jb_fftw_tde_result_hpp
+
+#include <jb/detail/array_traits.hpp>
+#include <jb/fftw/aligned_multi_array.hpp>
+
+namespace jb {
+namespace fftw {
+
+/**
+ * tde_result is a type to handle the result of operating over the
+ * last dimension array of a multi array of timeseries.
+ * This is used when the operation returns a value out of the array,
+ * therefore tde_result has one dimension less than the
+ * tde_result constructor's argument.
+ */
+
+/**
+ * Handles TDE result for container type timeseries.
+ *
+ * @tparam container_t container type timeseries
```

Comment:
Reminds me of the standard X11 joke:

```
Display display; // the display
```

I kind of know, tell me something that I did not know, for example:

```
 * @tparam container_t the representation for a timeseries, expected to be a one-dimensional container such as std::vector<> or std::deque<>
```